In [2]:
import pandas as pd
import numpy as np

### Get TRY plant ids for species of interest
This is for querying the TRY database

In [5]:
## A try species list
try_spp = "../try_sp_list.csv"
try_spp_df = pd.read_csv(try_spp, encoding_errors="ignore")

/tmp/ipykernel_9984/744311194.py:3: DtypeWarning: Columns (3,4,5,6) have mixed types. Specify dtype option on import or set low_memory=False.
  try_spp_df = pd.read_csv(try_spp, encoding_errors="ignore")


In [7]:
target_shrubs = ['Arctostaphylos patula','Arctostaphylos manzanita',
                       'Ceanothus cordulatus','Ceanothus cuneatus',
                       'Ceanothus integerrimus','Ceanothus velutinus']

target_oaks = ['Quercus wislizeni','Quercus agrifolia','Quercus ilex','Quercus engelmannii','Quercus suber']

g_s = [oak.split(" ") for oak in target_oaks]

In [8]:
g_s

[['Quercus', 'wislizeni'],
 ['Quercus', 'agrifolia'],
 ['Quercus', 'ilex'],
 ['Quercus', 'engelmannii'],
 ['Quercus', 'suber']]

In [9]:
genus = np.array(g_s)[:,0]
species = np.array(g_s)[:,1]
species

array(['wislizeni', 'agrifolia', 'ilex', 'engelmannii', 'suber'],
      dtype='<U11')

In [10]:
genus = np.array(g_s)[:,0]
species = np.array(g_s)[:,1]

sp_id = []
i = 0
for g,s in zip(genus,species):
    i = i + 1
    sp_filter = (try_spp_df['AccSpeciesName'] == g) & (try_spp_df['chrysotricha'] == s)
    sp_id.append(list(try_spp_df.loc[sp_filter]["AccSpeciesID"]))

In [12]:
sp_filter

0         False
1         False
2         False
3         False
4         False
          ...  
279870    False
279871    False
279872    False
279873    False
279874    False
Length: 279875, dtype: bool

In [11]:
sp_id

[[45504, 374177], [45314], [45402], [45370], [45490]]

PIPO: 42168,251255
ABCO: 39
CADE: 9622

Target shrubs:4520, 4513, 204236, 204237, 204238, 281021, 281022, 281023, 11291, 11293, 212524, 212525, 212526, 11308, 11331, 295294,

QUCH, QUKE, QUDO, QUAG, QUEN, QUWI
45408, 45346, 374041, 45363, 45314, 45370, 45504, 374177

Data to get

VCmax (186,185) is available or blue oak (45363)
Xylem hydraulic vulnerability curve (P20, P50, P80) (3479,719) for QUAG
sla and leaf nitrogen 

11, 3115, 3116, 3117, 3085, 3086, 3106, 3107, 570, 50, 24, 153, 12, 52, 6, 4, 3479, 719, 171, 186, 185, 487, 537, 186,185,45363


4520],
 [4513, 204236, 204237, 204238, 281021, 281022, 281023],
 [11291],
 [11293, 212524, 212525, 212526],
 [11308],
 [11331, 295294